master/internal/api_command.go

```diff
@@ -103,17 +103,18 @@ func (a *apiServer) getCommandLaunchParams(ctx context.Context, req *protoComman
 		return nil, launchWarnings, fmt.Errorf("checking respurce availability: %v", err.Error())
 	}
 	// Get the base TaskSpec.
-	taskContainerDefaults := a.m.getTaskContainerDefaults(poolName)
+	taskContainerDefaults := a.m.rm.TaskContainerDefaults(
+		a.m.system,
+		resources.ResourcePool,
+		a.m.config.TaskContainerDefaults,
```

Comment:
maybe ideally this wouldn't get passed, but i guess i need to think about the abstractions some more